### ISSUE TITLE
The code and software icp results are inconsistent

### ISSUE THREAD

**Issue by darkdelting** [CloudCompare/CCCoreLib#115]
software:
![image](https://github.com/user-attachments/assets/8fb999a2-d29e-4455-a3d4-29a54181e8b1)
![image](https://github.com/user-attachments/assets/0a599429-f47e-44bb-8679-bb945caa0ba5)
![image](https://github.com/user-attachments/assets/11b05f8e-3f31-4311-9f32-d95316076a08)

code:
![image](https://github.com/user-attachments/assets/0cf1aab0-6455-42ba-9853-3afbd8ac0a5a)
![image](https://github.com/user-attachments/assets/13a6b316-99d9-4601-b1b6-e0413b922a6f)
![image](https://github.com/user-attachments/assets/18706f90-f1fa-4ad0-a997-38a147e58a25)

**Comment by dgirardeau** [comment]
Sorry, I am not sure to fully understand the issue? Do you mean that the default parameters of the CCCoreLib library are different from the CloudCompare GUI version? (in which case that's definitely possible).

**Comment by darkdelting** [comment]
> Sorry, I am not sure to fully understand the issue? Do you mean that the default parameters of the CCCoreLib library are different from the CloudCompare GUI version? (in which case that's definitely possible).

Yes , I use the CloudCompare v2.13.2 and I'm not sure if it's the problem I'm using.

**Comment by dgirardeau** [comment]
Well, CCCoreLib and CloudCompare are 2 different projects. They don't need to use the same default parameters.

**Comment by darkdelting** [comment]
> Well, CCCoreLib and CloudCompare are 2 different projects. They don't need to use the same default parameters.

You mean ,  That is CCCoreLib and CloudCompare are 2 different projects and use the same algorithm, but default parameters is different ? what is CloudCompare default parameters . Can I align the performance of the two algorithms?

**Comment by dgirardeau** [comment]
You can see how CloudCompare calls CCCoreLib here: 
- https://github.com/CloudCompare/CloudCompare/blob/master/qCC/mainwindow.cpp#L3798
- https://github.com/CloudCompare/CloudCompare/blob/master/qCC/ccRegistrationTools.cpp#L46

Most of the parameters are set via the GUI, so it may not be simple to find the default values easily (you can also look at the dialog's ui file).

**Comment by darkdelting** [comment]
Thanks for your anwser!! As you can see from the code, the parameters have been recalculated in the backend ，which is diffrent from ui  
![image](https://github.com/user-attachments/assets/ab05f95a-114e-426b-83f1-31178affdb5c)
![image](https://github.com/user-attachments/assets/3bc1bf7a-a07e-4bb2-9c16-82b99311840b)
![image](https://github.com/user-attachments/assets/b134ee49-e59c-4003-8fd9-dc3d4cc40cd0)

**Comment by dgirardeau** [comment]
Well, 'baseIncrement' is just for memory allocation... We then simply update the overlap ratio to only consider points within the 'max search distance' (if any). Because the UI let's you play with various conflicting concepts.

**Comment by cama36** [comment]
Hi, I have tested the effect of ICP based on PCL, which is much worse than that of CCICP. In most cases, CCICP works well. So now I want to use CCICP in my own PCL-based project. Can you provide a minimum example? This is very important to me. Thanks.
